src/components/button.tsx

```diff
@@ -1,14 +1,17 @@
-import { ComponentPropsWithoutRef, PropsWithChildren } from "react";
+import {
+  ComponentPropsWithoutRef,
+  MouseEventHandler,
+  PropsWithChildren,
+} from "react";
 
 import { cn } from "../lib/utils";
 import { Loader } from "./loader";
 
-interface ButtonProps
-  extends ComponentPropsWithoutRef<"button">,
-    PropsWithChildren {
-  onClick: () => void;
-  loading?: boolean;
-}
+type ButtonProps = ComponentPropsWithoutRef<"button"> &
+  PropsWithChildren & {
+    onClick: MouseEventHandler<HTMLButtonElement>;
+    loading?: boolean;
+  };
```

Comment:
button already has children & onclick 
```suggestion
type ButtonProps = ComponentPropsWithoutRef<"button"> & {
    loading?: boolean;
  };
```

Reply:
Thanks, you're right! I wanted to make the onClick prop mandatory for this component. What’s the best way to achieve that while keeping the existing props? I appreciate your guidance!

Reply:
Then yes you can overwrite it as you did (just remove the PropsWithChildren because thats already there).
But, on that note, what if you want to use this button as the default submit button of a form? It doesn't need an onclick handler there, is it wise to overwrite HTML's native attributes?

Reply:
Absolutely not. I’ll remove PropsWithChildren and ensure the button supports use as a default submit button without overwriting native attributes. Thanks!